webapp/channels/src/utils/performance_telemetry/reporter.ts

```diff
@@ -78,13 +85,17 @@ export default class PerformanceReporter {
     protected reportPeriodBase = 60 * 1000;
     protected reportPeriodJitter = 15 * 1000;
 
-    constructor(client: Client4, store: Store<GlobalState>) {
+    constructor(client: Client4, store: Store<GlobalState>, desktopAPI: DesktopAppAPI) {
         this.client = client;
         this.store = store;
+        this.desktopAPI = desktopAPI;
 
         this.platformLabel = getPlatformLabel();
         this.userAgentLabel = getUserAgentLabel();
 
+        // We want to submit by prerelease version if it exists, so we don't muddy up the metrics for the release builds
+        this.desktopAppVersion = desktopAPI.getPrereleaseVersion()?.split('.')[0] ?? desktopAPI.getAppVersion() ?? 'unknown';
```

Comment:
What does the version number here look like in the case of a prerelease build? I'm trying to understand why we need to split here instead of just using the raw prerelease version value.

I'm also a bit leery of including that since I wouldn't expect it to change very often between release candidates (assuming that's what prerelease means in this case), but I'm not particularly against it

Reply:
Also, would you mind moving this to `./platform_detection` for consistency?

Reply:
Prerelease builds in this case will refer to `rc`, `nightly` and `develop`. For `rc` I can see a reason to add the actual release number but it's not as important, but for `nightly` and `develop` those numbers will change frequently and I don't want to have that many possible values around for that many different versions, most of which will not be used by end users.